build.override.targets

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0"  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--if ServiceUri is set, the service has been already started on another machine-->
+  <Target Name="WcfSetup" Condition="($(WithCategories.Contains('OuterLoop')) or '$(OuterLoop)' == 'true') and ('$(ServiceUri)' == '') and $(OS.Contains('Windows'))" BeforeTargets="BuildAllProjects">
```

Comment:
You asked a great question about whether BuildAllProjects is the right "before" target.  Another possibility is using the mechanism already being used by dir.traversal.targets, something like this:
```
  <PropertyGroup>
    <TraversalBuildDependsOn>
      WcfSetup;
      $(TraversalBuildDependsOn);
    </TraversalBuildDependsOn>
<PropertyGroup>
```
Another possibility is to say "Test" is the target you run before (to be consistent with being the target you run after) and to set an MsBuild property in this file to indicate "I have been called before".  If you did that, I think it would give a better experience for the /t:BuildAndTest that some of us use to run just one test project at a time.

Let's not block on the decision here but consider what we might do after the merge.

/cc @weshaggard for thoughts about where/how is the right place to hook .